client/packages/cli/src/index.js

```diff
@@ -412,7 +413,37 @@ program
 program.parse(process.argv);
 
 // command actions
+async function checkVersion() {
+  const resp = await fetchJson({
+    method: 'GET',
+    path: `/dash/cli/version`,
+    debugName: 'CLI version',
+    errorMessage: 'Failed to fetch CLI version',
+  });
+
+  if (!resp.ok) {
+    console.log('Warning! Failed to check backend version');
+    process.exit(1);
+  }
+
+  const cliVersion = version.startsWith('v') ? version.substring(1) : version;
+  const minVersionMap = resp.data['min-version'];
+
+  if (minVersionMap) {
+    const { major, minor, patch } = minVersionMap;
+    const minVersion =
+      `${major}.${minor}.${patch}` + (minVersionMap['dev?'] ? '-dev' : '');
+    if (semver.lt(cliVersion, minVersion)) {
+      console.log(
+        `${chalk.red('ERROR')} Your CLI version ${chalk.red(cliVersion)} is too old to use with Instant backend, requires ${chalk.red(minVersion)} or later.`,
+      );
+      process.exit(1);
+    }
+  }
+}
+
 async function handlePush(bag, opts) {
+  await checkVersion();
```

Comment:
I worry about the extra latency here, but probably not too big a deal. We can see how it rolls 🫡